### PR TITLE
with-proxy: create demo with a proxy

### PR DESCRIPTION
**Pull request by jdrouet**
Signed-off-by: Jérémie Drouet <jeremie.drouet@datadoghq.com>

### CODE REVIEW

File: with-proxy/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+  nginx:
+    image: nginx
+    volumes:
+      - ./nginx:/etc/nginx:ro
+    networks:
+      - private
+
+  squid:
+    image: babim/squid
```
Comment (jszwedko) [inline with-proxy/docker-compose.yml:13]: It doesn't really matter, but I did discover Datadog publishes a Squid image too if you wanted to use that one: https://hub.docker.com/r/datadog/squid/

File: with-proxy/vector/vector.toml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# can be configured globally
```
Comment (jszwedko) [inline with-proxy/vector/vector.toml:1]: 👍 I like these notes you've left (here and for the env var support).

File: with-proxy/readme.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+To start it, you first need to clone this repository, go into the `with-proxy` folder and run `docker-compose up`.
+
+## What it does.
+
+The [`docker-compose.yml`](./docker-compose.yml) declares 2 times of network. The `public` network to which only Vector and Squid have access, and the `private` to which only Nginx and Squid have access. Vector and Nginx can't communicate with each other.
+
+To allow Vector to reach Nginx, the proxy configuration has to be set on Vector. It can be done in 3 different ways:
+- with an environment variable, by setting `HTTP_PROXY=http://squid:3128`, see the comment in [`docker-compose.yml`](./docker-compose.yml).
+- by setting the proxy globaly, like in the first three lines of [`vector/vector.toml`](./vector/vector.toml).
+- by setting the proxy at the component level, in the section `[sources.nginx.proxy]` of [`vector/vector.toml`](./vector/vector.toml).
```
Comment (jszwedko) [inline with-proxy/readme.md:16]: 👍

File: with-proxy/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    networks:
+      - public
+      - private
+
+  vector:
+    image: timberio/vector:nightly-alpine
+    volumes:
+      - ./vector/vector.toml:/etc/vector/vector.toml:ro
+      - ./logs:/tmp/logs
```
Comment (jszwedko) [inline with-proxy/docker-compose.yml:22]: Do we need this directory mounted? I'm not seeing anything using it.

**Review by jszwedko — APPROVED** [review]
Nice! I'm a fan of these examples.

File: with-proxy/docker-compose.yml (hunk 1 above)
Comment (jdrouet) [reply with-proxy/docker-compose.yml:13]: Yeah, I found it as well, ask people if it's still maintained but apparently it was just a test that they did 🤷 So better not use it I think.

File: with-proxy/docker-compose.yml (hunk 4 above)
Comment (jdrouet) [reply with-proxy/docker-compose.yml:22]: ah good point, a copy/paste that went too fast...

File: with-proxy/docker-compose.yml (hunk 1 above)
Comment (jszwedko) [reply with-proxy/docker-compose.yml:13]: Aha 😄 Agreed then.

**Pull request merged**